src/wp-admin/theme-editor.php

```diff
@@ -212,19 +212,44 @@
 	);
 }
 
-if ( preg_match( '/\.css$/', $file ) && ! wp_is_block_theme() && current_user_can( 'customize' ) ) {
-	$message = '<p><strong>' . __( 'Did you know?' ) . '</strong></p><p>' . sprintf(
-		/* translators: %s: Link to Custom CSS section in the Customizer. */
-		__( 'There is no need to change your CSS here &mdash; you can edit and live preview CSS changes in the <a href="%s">built-in CSS editor</a>.' ),
-		esc_url( add_query_arg( 'autofocus[section]', 'custom_css', admin_url( 'customize.php' ) ) )
-	) . '</p>';
-	wp_admin_notice(
-		$message,
-		array(
-			'type' => 'info',
-			'id'   => 'message',
-		)
-	);
+if ( preg_match( '/\.css$/', $file ) ) {
+	if ( ! wp_is_block_theme() && current_user_can( 'customize' ) ) {
+		$message = '<p><strong>' . __( 'Did you know?' ) . '</strong></p><p>' . sprintf(
+			/* translators: %s: Link to Custom CSS section in the Customizer. */
+			__( 'There is no need to change your CSS here &mdash; you can edit and live preview CSS changes in the <a href="%s">built-in CSS editor</a>.' ),
+			esc_url( add_query_arg( 'autofocus[section]', 'custom_css', admin_url( 'customize.php' ) ) )
+		) . '</p>';
+		wp_admin_notice(
+			$message,
+			array(
+				'type' => 'info',
+				'id'   => 'message',
+			)
+		);
+	} elseif ( wp_is_block_theme() && current_user_can( 'edit_theme_options' ) ) {
+		$message = '<p><strong>' . __( 'Did you know?' ) . '</strong></p><p>' . sprintf(
+			/* translators: %s: Link to Additional CSS section in the Site Editor. */
+			__( 'There is no need to change your CSS here &mdash; you can edit and live preview CSS changes in the <a href="%s">built-in CSS editor</a>.' ),
+			esc_url( admin_url( 'site-editor.php?p=%2Fstyles&section=%2Fcss' ) )
+		) . '</p>';
+		wp_admin_notice(
+			$message,
+			array(
+				'type' => 'info',
+				'id'   => 'message',
+			)
+		);
+	}
+	if ( file_exists( preg_replace( '/\.css$/', '.min.css', $file ) ) ) {
+		$message = '<p><strong>' . __( 'There is a minified version of this stylesheet.' ) . '</strong></p><p>' .
+			__( 'It is likely that this unminified stylesheet will not be served to visitors.' ) . '</p>';
+		wp_admin_notice(
+			$message,
+			array(
+				'type' => 'warning',
```

Comment:
This will allow plugins to filter the notice using `wp_admin_notice_markup`, for example a plugin that generates the minimized version server side.


```suggestion
				'type' => 'warning',
				'id'   => 'wp-css-min-warning',
```